plugins/accelerated-peft/tests/test_gptqmodel.py

```diff
@@ -248,11 +248,15 @@ def test_quantizing_pretrained_model_outputs_match(
         refactored_logits = refactored_model(input_ids).logits
 
     # Measure the distribution error with KD Loss
+    # flatten as a single batch bs*seqlen
+    # since batchmean sums the loss and averages on dim=0
     loss_fn = torch.nn.KLDivLoss(reduction="batchmean")
     # input should be a distribution in the log space
-    input = torch.nn.functional.log_softmax(refactored_logits, dim=1)
+    input = torch.nn.functional.log_softmax(refactored_logits, dim=-1)
+    input = torch.flatten(input, start_dim=0, end_dim=1)
     # target must be prob distribution
-    target = torch.nn.functional.softmax(original_logits, dim=1)
+    target = torch.nn.functional.softmax(original_logits, dim=-1)
+    target = torch.flatten(target, start_dim=0, end_dim=1)
```

Comment:
if after you flatten it becomes a 1-dim tensor why dont you just do `reduction=sum`

also i think the canoical way to flatten is `target.view(...,-1)`